### PR TITLE
Feature/approval date validation

### PR DESCRIPTION
**Pull request by JB12298**

### CODE REVIEW

File: src/EA.Weee.Web/Areas/Admin/Controllers/AddAatfController.cs [hunk 1]
```diff
@@ -37,13 +35,20 @@ public class AddAatfController : AdminController
...
-
-        public AddAatfController(ISearcher<OrganisationSearchResult> organisationSearcher, Func<IWeeeClient> apiClient, BreadcrumbService breadcrumb, IWeeeCache cache)
+        private readonly IAatfViewModelValidatorWrapper validator;
+
+        public AddAatfController(
+            ISearcher<OrganisationSearchResult> organisationSearcher,
+            Func<IWeeeClient> apiClient,
+            BreadcrumbService breadcrumb,
+            IWeeeCache cache,
+            IAatfViewModelValidatorWrapper validator)
```
Comment (JohnPadgettCivica) [inline src/EA.Weee.Web/Areas/Admin/Controllers/AddAatfController.cs]: Do we need need the validator wrapper in the controller if now attached to the view model

File: src/EA.Weee.Web.Tests.Unit/Areas/Admin/Validation/AatfViewModelValidatorTests.cs [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+        public void RuleFor_ApprovalDateOutOfBounds_ErrorShouldOccur(string input)
+        {
+            var model = GenerateViewModel();
+            model.ApprovalDate = Convert.ToDateTime(input);
+
+            validator = new AatfViewModelValidator();
+            var validationResult = validator.Validate(model);
+
+            validationResult.IsValid.Should().BeFalse();
+            validationResult.Errors.Count.Should().Be(1);
```
Comment (JohnPadgettCivica) [inline src/EA.Weee.Web.Tests.Unit/Areas/Admin/Validation/AatfViewModelValidatorTests.cs]: Add check that the message is correct

File: src/EA.Weee.Web/Areas/Admin/Controllers/AddAatfController.cs (hunk 1 above)
Comment (JB12298) [reply src/EA.Weee.Web/Areas/Admin/Controllers/AddAatfController.cs]: Removed

File: src/EA.Weee.Web.Tests.Unit/Areas/Admin/Validation/AatfViewModelValidatorTests.cs (hunk 2 above)
Comment (JB12298) [reply src/EA.Weee.Web.Tests.Unit/Areas/Admin/Validation/AatfViewModelValidatorTests.cs]: Added

**Pull request merged**